client/components/banner/test/index.jsx

```diff
@@ -1,165 +1,197 @@
-		expect( comp.find( 'TrackComponentView' ) ).toHaveLength( 1 );
+		renderWithProvider( <Banner { ...props } event="test" /> );
+		expect( recordTracksEvent.mock.calls[ 0 ][ 1 ].cta_name ).toBe( 'test' );
 	} );
 
 	test( 'should not record Tracks event when event is not specified', () => {
-		const comp = shallow( <Banner { ...props } /> );
-		expect( comp.find( 'TrackComponentView' ) ).toHaveLength( 0 );
+		renderWithProvider( <Banner { ...props } /> );
+		expect( recordTracksEvent ).not.toHaveBeenCalled();
 	} );
 
 	test( 'should render Card with href if href prop is passed', () => {
-		const comp = shallow( <Banner { ...props } href={ '/' } /> );
-		expect( comp.find( Card ) ).toHaveLength( 1 );
-		expect( comp.find( Card ).props().href ).toBe( '/' );
+		const { container } = render( <Banner { ...props } href={ '/' } /> );
+		expect( container.firstChild ).toHaveClass( 'card' );
+		expect( container.firstChild ).toHaveAttribute( 'href', '/' );
 	} );
 
 	test( 'should render Card with no href if href prop is passed but disableHref is true', () => {
-		const comp = shallow( <Banner { ...props } href={ '/' } disableHref={ true } /> );
-		expect( comp.find( Card ) ).toHaveLength( 1 );
-		expect( comp.find( Card ).props().href ).toBeNull();
+		const { container } = render( <Banner { ...props } href={ '/' } disableHref={ true } /> );
+		expect( container.firstChild ).toHaveClass( 'card' );
+		expect( container.firstChild ).not.toHaveAttribute( 'href', '/' );
 	} );
 
 	test( 'should render Card with href if href prop is passed but disableHref is true and forceHref is true', () => {
-		const comp = shallow(
+		const { container } = render(
 			<Banner { ...props } href={ '/' } disableHref={ true } forceHref={ true } />
 		);
-		expect( comp.find( Card ) ).toHaveLength( 1 );
-		expect( comp.find( Card ).props().href ).toBe( '/' );
+		expect( container.firstChild ).toHaveClass( 'card' );
+		expect( container.firstChild ).toHaveAttribute( 'href', '/' );
 	} );
 
-	test( 'should render Card with no href and CTA button with href if href prop is passed and callToAction is also passed', () => {
-		const comp = shallow( <Banner { ...props } href={ '/' } callToAction="Go WordPress!" /> );
-		expect( comp.find( Card ) ).toHaveLength( 1 );
-		expect( comp.find( Card ).props().href ).toBeNull();
-		expect( comp.find( Card ).props().onClick ).toBeNull();
+	test( 'should render Card with no href and CTA button with href if href prop is passed and callToAction is also passed', async () => {
+		const handleClick = jest.fn();
+		const { container } = render(
+			<Banner { ...props } href={ '/' } callToAction="Go WordPress!" onClick={ handleClick } />
+		);
+
+		await userEvent.click( container.firstChild );
+		await userEvent.click( screen.getByText( 'Go WordPress!' ) );
+
+		expect( handleClick ).toHaveBeenCalledTimes( 1 );
+
+		expect( container.firstChild ).toHaveClass( 'card' );
+		expect( container.firstChild.href ).toBeUndefined();
```

Comment:
This makes sense to be undefined, there's no such way to access props with testing library 😉  Let's try to look for the representation of this in the UI and test against that instead.